compiler/luci-interpreter/src/kernels/Conv2D.cpp

```diff
@@ -118,9 +118,17 @@ void Conv2D::configure()
     input()->element_type() != DataType::S16 && (need_dilated_im2col || need_non_dilated_im2col);
   if (_need_im2col)
   {
-    const int input_depth = input_shape.dim(3);
-    Shape im2col_shape{batches, output_height, output_width,
-                       input_depth * filter_height * filter_width};
+    tflite::ConvParams params{};
+    params.padding_values.height = _padding_height;
+    params.padding_values.width = _padding_width;
+    params.stride_height = _params.stride_height;
+    params.stride_width = _params.stride_width;
+    params.dilation_height_factor = _params.dilation_height_factor;
+    params.dilation_width_factor = _params.dilation_width_factor;
+
+    Shape im2col_shape = luci_interpreter_pal::TemporaryTensorShape(
+      params, getTensorShape(input()), getTensorShape(filter()), getTensorShape(output()));
+
```

Comment:
Maybe rename `im2col` to `scratchpad` and move this logic to PAL?
Rationale is following:
- We do not need this buffer for `mcu` PAL at all
- This is not `im2col` tensor in `cmsis` PAL, it uses more memory efficient approach in MCU kernel

Reply:
Done